### DISCUSSION TITLE
Tree Sitter for parsing

### DISCUSSION THREAD

**Discussion by jordanbrauer** [webonyx/graphql-php#890] [Ideas]
Has the idea of using [Tree Sitter](https://tree-sitter.github.io/tree-sitter) for parsing been tossed around yet? I know it does not have PHP [bindings](https://tree-sitter.github.io/tree-sitter/#language-bindings) (though it has Ruby, and Node, so it should be possible), or an official GraphQL grammar, but if these two things existed (yay OSS), it might be neat to see the difference in performance when parsing schema documents.

Thoughts?

> **Note:** excuse me if this actually doesn't make any sense :p

**Comment by aszenz** [comment]
Interesting idea, how will the native dependency on tree sitter be provided by the package

**Reply by jordanbrauer** [reply]
Not sure; haven't really thought about it much beyond the initial idea. I guess I imagine that it would be an opt-in experience, "Install _this_ PHP extension to use Tree Sitter parsing".
